storage/remote/queue_manager.go

```diff
@@ -428,11 +476,11 @@ func (s *shards) stop(deadline time.Duration) {
 	return
 }
 
-func (s *shards) enqueue(sample *model.Sample) bool {
+func (s *shards) enqueue(sample tsdb.RefSample) bool {
 	s.qm.samplesIn.incr(1)
 
-	fp := sample.Metric.FastFingerprint()
-	shard := uint64(fp) % uint64(len(s.queues))
+	// fp := sample.Metric.FastFingerprint()
+	shard := uint64(sample.Ref) % uint64(len(s.queues))
```

Comment:
How is the ref generated?  Is in only ever incrementing?  I just wonder if this will balance across queues correctly.

Reply:
the series reference id's are atomically incremented: https://github.com/prometheus/tsdb/blob/master/head.go#L1109-L1110